api/errors/errors.go

```diff
@@ -1,47 +1,57 @@
 package errors
 
-import "net/http"
+// NotFoundError signals that the requested object doesn't exist
+type NotFoundError interface {
+	error
+	NotFound()
+}
 
-// apiError is an error wrapper that also
-// holds information about response status codes.
-type apiError struct {
+// InvalidParameterError signals that the user input is invalid
+type InvalidParameterError interface {
```

Comment:
I think it might be good to still keep these interfaces exported so it shows up in the package docs. But also +1 to adding the `Is<SomeErr>` methods.